### PR TITLE
Fix argument validation for WebHeaderCollection.Add

### LINKED ISSUE #20164: [porting bug] WebHeaderCollection.Add(null, "value") throws NRE in netcoreapp, ArgumentNullException in netfx`
```cs
[Fact]
public void Add_NullName_ThrowsArgumentNullException()
{
    var headers = new WebHeaderCollection();
    Assert.Throws<ArgumentNullException>("name", () => headers.Add(null, "value"));
}
```

This causes some HttpListener tests to fail after I've fixed #20161 

I have a fix and will submit a PR

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #20164

Tests pass with netfx and netcoreapp

### CODE REVIEW

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs [hunk 1]
```diff
@@ -404,8 +430,107 @@ public static IEnumerable<object[]> SerializeDeserialize_Roundtrip_MemberData()
...
         }
 
+        public static IEnumerable<object[]> Add_Value_TestData()
+        {
+            yield return new object[] { null, string.Empty };
+            yield return new object[] { string.Empty, string.Empty };
+            yield return new object[] { "VaLue", "VaLue" };
+            yield return new object[] { "  value  ", "value" };
+
+            // Documentation says this should fail but it does not.
```
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: If you haven't already, can you open an issue for a doc bug?

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs [hunk 2]
```diff
@@ -503,20 +628,6 @@ public void HttpRequestHeader_AddHeader_Failure()
         }
 
         [Fact]
-        public void HttpRequestHeader_AddMissingColon_Failure()
-        {
-            WebHeaderCollection w = new WebHeaderCollection();
-            Assert.Throws<ArgumentException>(() => w.Add("ContentType#text/html"));
```
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: What replaced this?

Comment (hughbe) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: https://github.com/dotnet/corefx/pull/20169/files/cacfcee42bd482232853f5493544c8ec46177794#diff-5a05d6ae9e2152bd727773617a68e165R522

[InlineData("nocolon", "header")]

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: Opened https://github.com/dotnet/corefx/issues/20180

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop netfx Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test outerloop netcoreapp Ubuntu16.10 Debug

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: Ok

File: src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.WebHeaderCollection/tests/WebHeaderCollectionTest.cs]: Thanks

**Comment by stephentoub** [comment]
@karelz, we should port this to 2.0.

**Comment by karelz** [comment]
Agreed. It is already tracked by #20164, I will add it to the uber-porting issue once it is merged.

**Pull request merged**